src/ua.c

```diff
@@ -932,6 +932,46 @@ int ua_uri_complete(struct ua *ua, struct mbuf *buf, const char *uri)
 }
 
 
+static bool uri_host_local(const struct uri *uri)
+{
+	const struct sa *sal;
+	struct sa sap;
+	int err;
+
+	if (!uri)
+		return false;
+
+	if (!pl_strcmp(&uri->host, "localhost"))
+		return true;
+
+	if (!pl_strcmp(&uri->host, "127.0.0.1"))
+		return true;
+
+	if (!pl_strcmp(&uri->host, "::1"))
+		return true;
+
+	sal = net_laddr_af(baresip_network(), AF_INET);
+
+	err = sa_set(&sap, &uri->host, 0);
+	if (err)
+		return false;
+
+	if (sal && sa_cmp(sal, &sap, SA_ADDR))
+		return true;
+
+#ifdef HAVE_INET6
+	sal = net_laddr_af(baresip_network(), AF_INET6);
+	if (sal && sa_cmp(sal, &sap, SA_ADDR))
+		return true;
+#endif
+
+	return false;
+}
```

Comment:
this code looks fine, but it is also possible to make an array of hosts/afs and compare it.

example code:

```c
static bool uri_host_local(const struct uri *uri)
{
	const char *hostv[] = {
		"localhost",
		"127.0.0.1",
		"::1"
	};
	int afv[2] = {AF_INET, AF_INET6};
	const struct sa *sal;
	struct sa sap;
	size_t i;
	int err;

	if (!uri)
		return false;

	for (i=0; i<ARRAY_SIZE(hostv); i++) {

		if (!pl_strcmp(&uri->host, hostv[i]))
			return true;
	}

	for (i=0; i<ARRAY_SIZE(afv); i++) {

		sal = net_laddr_af(baresip_network(), afv[i]);

		err = sa_set(&sap, &uri->host, 0);
		if (err)
			continue;

		if (sa_cmp(sal, &sap, SA_ADDR))
			return true;
	}

	return false;
}
```

it makes it easier in the future to add new host types should it
be needed, for example "127.0.0.2"